### ISSUE TITLE
Error: Adapter registry error using typescript

### ISSUE THREAD

**Issue by leandrodaf** [socketio/socket.io-redis-adapter#452]
Problem registering adapter in socketIo using typescript.

![image](https://user-images.githubusercontent.com/12107458/165299539-9615acd8-1ff1-4188-9489-5bd01cd7b1e1.png)

Packages

```json
{
    "devDependencies": {
       "typescript": "4.5.2"
    },
    "dependencies": {
       "@socket.io/redis-adapter": "^7.1.0",
       "socket.io": "^4.4.1"
    }
 }
```
------
I solved the problem by doing

![image](https://user-images.githubusercontent.com/12107458/165299376-5b51986d-5a8c-434b-ae73-39640b9aebae.png)

But it doesn't seem like the best way to me.

**Comment by aggarwaldev** [comment]
Pinning the socket.io version to 4.4.1 seems to fix this.

Add the following to your `package.json`:

` "socket.io": "4.4.1",`

**Comment by muratkemaldar** [comment]
Had this issue also the other day and just wanted to created an issue for this as well :) Was also using the same workaround with `as any`.

**Comment by jonathan-nielsen** [comment]
I'm getting the exact same issues by simple running `npm i -s socket.io @socket.io/redis-adapter redis` as the installation step in the documentation at https://socket.io/docs/v4/redis-adapter/ says. This gives me the following in my package.json.
```
"@socket.io/redis-adapter": "^7.1.0"
"socket.io": "^4.5.0"
"redis": "^4.0.6"
```

**Comment by filoscoder** [comment]
Same issue here, `as any` solve it for now... expecting type upgrades soon 🥲

**Comment by darrachequesne** [comment]
I'm looking into it, it seems to be linked to the features included in `socket.io@4.5.0`.

**Comment by schw4rzlicht** [comment]
I think I have narrowed down the issue to this commit and the typing changes that came with it: https://github.com/tough-solutions/socket.io/commit/891b1870e92d1ec38910f03bb839817e2d6be65a

**Comment by darrachequesne** [comment]
This should be fixed by https://github.com/socketio/socket.io-redis-adapter/commit/e4c40cc8a9ad8803f03bcbbfd6b713f3c082ee28, included in version `7.2.0`. Thanks for the heads-up!

**Comment by muratkemaldar** [comment]
Can confirm, error is gone.
